### ISSUE TITLE
There are bugs present when using YAML to directly convert to ONNX.

### ISSUE THREAD

**Issue by HiiHongFe** [ultralytics/ultralytics#3293]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Export

### Bug

I used a YAML file to directly initialize the model and convert it to ONNX, but during initialization, it notifies me that the "kpt_shape" parameter is missing. After investigating the issue, it seems that the initialization was not done properly. I suggest modifying the position in "ultralytics-main/ultralytics/nn/tasks.py".

My Code：
![image](https://github.com/ultralytics/ultralytics/assets/44940083/4d3bbbe9-9941-4e88-a12b-a04b287abba8)

Error：
![image](https://github.com/ultralytics/ultralytics/assets/44940083/c54e94fc-f89a-4d91-8b81-f9c532040f33)
Fix:
![image](https://github.com/ultralytics/ultralytics/assets/44940083/01f3c0fb-a99d-4c56-81f3-67c41c6baeee)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@HiiHongFe thanks for the bug report! I'm able to reproduce with this code. Added a TODO to queue for a fix.

```python
from ultralytics import YOLO

YOLO('yolov8n-pose.yaml').export(format='onnx')
```

**Comment by glenn-jocher** [comment]
@HiiHongFe great news 😃! Your original issue may now be resolved ✅ in PR #3312. 

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!
